### ISSUE TITLE
Using `pack`  - include project reference as dependency even if the id doesn't match?

### ISSUE THREAD

**Issue by ascjones** [fsprojects/Paket#670]
For a project template file, the rule for determining dependencies is (as per the [docs](http://fsprojects.github.io/Paket/template-files.html)):

`the output assemblies of any project references which do not have a matching template file`

However, in most of our projects the package id doesn't match the project name since we use a company prefix on the package names. This means that inter-project dependencies that also publish their own packages will be included in the lib folder but not as dependencies.

So, how can I make it so that our inter project dependencies are added as dependencues even if the id doesn't match?

/cc @mavnn

**Comment by mavnn** [comment]
It's not done on id at all: it's based on whether the linked proj file has
an associated paket.template.

Is it not working?

**Comment by ascjones** [comment]
Oh right, let me try again

**Comment by ascjones** [comment]
It is the behaviour I'm getting (v0.31.6.0):

paket.template id in referenced project == project name: project reference included as dependency
paket.tempate id in referenced project != project name: project reference included in lib folder

Quick look at the code - is this it:

https://github.com/fsprojects/Paket/blob/016da709ba8a74c7244bc9fbba3bac4be05adf8f/src/Paket.Core/PackageMetaData.fs#L145

Would run the debugger but gotta run now. Will check back tomorrow.

**Comment by mavnn** [comment]
Ah, no: it's here https://github.com/fsprojects/Paket/blob/016da709ba8a74c7244bc9fbba3bac4be05adf8f/src/Paket.Core/PackageProcess.fs#L83 . id should be the project name (which is how we look it up later) not the nupkg id (which is what it's being set to currently).

**Comment by forki** [comment]
So it's a bug?

**Comment by mavnn** [comment]
Yes

**Comment by ascjones** [comment]
Okay I need this so I'll fix it now, unless @mavnn is already on it

**Comment by mavnn** [comment]
Go for it - I'm on other stuff today
